extensions/shex-map/lib/extension-utils.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * A file with common utility functions used by the extensions.
+ */
+
+var _ = require('underscore');
+
+/**
+ * Given a variable name, looks up its prefix, and  replacing the shorthand 
+ * prefix name in the variable with l name.
+ * 
+ * @param varName variable name to expand
+ * @param prefixes a list of the prefix short name/full name mappings
+ *
+ * @return the fully expanded name
+ */
+function applyPrefix(varName, prefixes) {
+
+  var onURIPattern = varName.match(/^ *(?:<([^>]*)>|([^:]*):([^ ]*)) *$/);
+  if (!onURIPattern)
+    throw Error("variable \"" + varName + "\" didn't match expected pattern!");
+
+  var expandedVarName;
+
+  if (onURIPattern[1])
+    // explicit URL
+    expandedVarName = onURIPattern[1];
+
+  else if (onURIPattern[2] in prefixes)
+    // prefixed URL
+    expandedVarName = prefixes[onURIPattern[2]] + onURIPattern[3];
+
+  else
+    // unknown prefix
+    throw Error("Unknown prefix " + onURIPattern[2] + " in \"" + varName + "\"!");
+
+  return expandedVarName;
+}
+
+function getVariable(p1, vars, prefixes) {
```

Comment:
Agree.  This API was originally Eric's.  I actually don't like the side-effect of updating expandedVars as well as returning the individual value, but I didn't want to make more extensive changes than I already did.    I'll make this change.